### ISSUE TITLE
FileSystemWatcher.Rename clips FullPath and Name

### ISSUE THREAD

**Issue by jrieken** [dotnet/corefx#6866]
* on OSX run the snippet below with a path to watch
* in that path run `mv existing.file newName.file`
* the rename event clips the first character of the new name

![screen shot 2016-03-14 at 16 07 39](https://cloud.githubusercontent.com/assets/1794099/13748859/ff9a6f1c-e9fe-11e5-9874-49194b4d05cd.png)

```c#
using System;
using System.IO;
using System.Threading.Tasks;

namespace ConsoleApplication
{

    public class Program
    {
        public static void Main(string[] args)
        {
            Watch(args[0]).Wait();
            Console.WriteLine("done");
        }

        private static async Task<string> Watch(string path)
        {
            FileSystemWatcher watcher = new FileSystemWatcher(path);

            var tcs = new TaskCompletionSource<string>();
            watcher.Renamed += (sender, arg) =>
            {
                Console.WriteLine(arg.ChangeType);
                Console.WriteLine(arg.OldFullPath);
                Console.WriteLine(arg.FullPath);
                Console.WriteLine(arg.Name);
            };

            watcher.EnableRaisingEvents = true;
            watcher.IncludeSubdirectories = true;

            Console.WriteLine("Waiting for a file to change...");
            var result = await tcs.Task;
            return result;
        }
    }
}

```

**Comment by jrieken** [comment]
from project.json `"System.IO.FileSystem.Watcher": "4.0.0-beta-23516"`

**Comment by sokket** [comment]
Yikes, that's not right. Looking now

**Comment by stephentoub** [comment]
@sokket, you were also working on adding tests that validated the paths were what we expected them to be, right?  Seems like a good thing to do now :wink:

**Comment by sokket** [comment]
@stephentoub Yeah, the next item on my list is to finish that change. Currently playing whack-a-bug with the builds change first, but I'll work on fixing this product issue in parallel. After these two, I'll get the path validation working

**Comment by sokket** [comment]
Fixed via #6873

**Comment by jrieken** [comment]
Thanks everyone
